accessors/dataflow/dataflow_accessor.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dataflowaccessor
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"cloud.google.com/go/dataflow/apiv1beta3/dataflowpb"
+	dataflowclient "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/dataflow"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"golang.org/x/exp/maps"
+)
+
+type DataflowAccessor interface {
+	// This function takes the template parameters (@parameters) and runtime environment config (@cfg) as input, and returns
+	// the generated jobId, equivalentGcloudCommand and error if any.
+	LaunchFlexTemplate(ctx context.Context, parameters map[string]string, cfg DataflowTuningConfig) (string, string, error)
+}
+
+type DataflowAccessorImpl struct{}
+
+func (dfA *DataflowAccessorImpl) LaunchFlexTemplate(ctx context.Context, parameters map[string]string, cfg DataflowTuningConfig) (string, string, error) {
```

Comment:
Can you please add coverage for this method?

https://screenshot.googleplex.com/6mXNkg3huwrxgPd

Reply:
Accessor methods like these calling third party methods cannot be mocked as we discussed. These would require some form integration testing using emulator for spanner accessor, but for dataflow accessor there is no good way. (without pushing the dataflowClient out and take that as an interface input, which goes against wanting to keep client management within accessors.)

Reply:
Thanks, I understand this better now. Two principles seem to be at odds with each other - wanting to keep the client management within accessors, and wanting to mock out clients to test the accessors. 

This is generally solved by injection. However, since we are not using injection, if I have to chose between the two, I would rather pass the client object by requesting it from `dataflow_client` and passing it to `dataflow_accessor` than leave this piece of code untested. 

I know there has been a bit of churn on this - we're figuring this out as we go - but I do think testing supersedes convention here.